### PR TITLE
Update Automated_Testing.md to add information about API testing

### PR DESCRIPTION
**Pull request by anudevgill**
This PR contains information on:

- The importance of API testing and its benefits
- Additional more complex benefits of API testing in large-scale projects
- Understanding the options for automated API testing in Postman
- How automated API testing can be set up through the Postman CLI and integrated into your CI/CD pipeline

### CODE REVIEW

**Review by RJAK11 — COMMENTED** [review]
I’ve reviewed your additions and your explanations are very clear and concise! Also including the step-by-step instructions about how to set up the tests is very helpful. 

One suggestion would be to add a brief explanation about whether or not API testing works the same if the requests require authorizations or credentials and what changes you would need to make if not.

**Comment by anudevgill** [comment]
That's a great point that I forgot to mention! I've now added those details about how API testing works when authorization is required. Is there anything else that you think I should change or perhaps anything that could be made more clear?

**Comment by RJAK11** [comment]
> That's a great point that I forgot to mention! I've now added those details about how API testing works when authorization is required. Is there anything else that you think I should change or perhaps anything that could be made more clear?

That looks great! Just one more minor suggestion: could you re-word the second-last statement in the second paragraph to specify which tests you are talking about exactly?

**Comment by chenikabukes** [comment]
This is a very thorough explanation of API testing which is very helpful. I have left some structural comments above and have one question remaining. With regards to authorisation, can you discuss the benefits or disadvantages of using refresh tokens and whether you would use them for API testing in Postman?

**Comment by anudevgill** [comment]
> That looks great! Just one more minor suggestion: could you re-word the second-last statement in the second paragraph to specify which tests you are talking about exactly?

Absolutely! I've made that change to make it more clear whether I'm referring to API testing generally or specifically in Postman. Does everything look good to merge now?

**Comment by anudevgill** [comment]
> This is a very thorough explanation of API testing which is very helpful. I have left some structural comments above and have one question remaining. With regards to authorisation, can you discuss the benefits or disadvantages of using refresh tokens and whether you would use them for API testing in Postman?

Thanks! I've added an explanation regarding the benefits of using refresh tokens/the form of scraping I described in my PR. For automated testing this really is a must so there aren't any clear disadvantages.

**Comment by chenikabukes** [comment]
> > This is a very thorough explanation of API testing which is very helpful. I have left some structural comments above and have one question remaining. With regards to authorisation, can you discuss the benefits or disadvantages of using refresh tokens and whether you would use them for API testing in Postman?
> 
> Thanks! I've added an explanation regarding the benefits of using refresh tokens/the form of scraping I described in my PR. For automated testing this really is a must so there aren't any clear disadvantages.
> 
> Does everything look good to merge now?

Thanks for adding these pieces! From my side this looks very comprehensive and clear now. Ready to merge!

**Comment by RJAK11** [comment]
> > This is a very thorough explanation of API testing which is very helpful. I have left some structural comments above and have one question remaining. With regards to authorisation, can you discuss the benefits or disadvantages of using refresh tokens and whether you would use them for API testing in Postman?
> 
> Thanks! I've added an explanation regarding the benefits of using refresh tokens/the form of scraping I described in my PR. For automated testing this really is a must so there aren't any clear disadvantages.
> 
> Does everything look good to merge now?

Yup, it looks good to me!

**Review by chenikabukes — APPROVED** [review]
This is ready to merge now!

File: Topics/Development_Process/Automated_Testing.md [hunk 1]
```diff
@@ -64,4 +64,18 @@ Integration testing, in contrast to unit testing, is meant to verify that your c
...
+
+#### API Testing
+
+Another important area of automated testing is API, or Application Program Interface, testing. API testing can be helpful to integrate into your automated tests as it ensures that the requests that your software sends are correctly received and in turn, you receive the right output. API testing is incredibly powerful and can be used to not only check that the right status is received but also to check that the data returned is of a particular form or has certain attributes, amongst other things. Errors returned in automated API tests can be a sign that there is an error in your code in the requests you send or that there has been a change to the API that you are accessing. Either way, these tests can be a great method of catching these issues early and resolving them before your code is deployed. In larger software systems, there are actually a number of applications of API testing beyond the two mentioned above which can include aspects such as penetration testing or security testing and a more extensive overview of these kinds of tests can be found at [https://blog.hubspot.com/website/api-testing](https://blog.hubspot.com/website/api-testing).
+
```
[the remark this answers is not in the document]
Comment (anudevgill) [reply Topics/Development_Process/Automated_Testing.md:72]: I reverted the PR and added these in!

File: Topics/Development_Process/Automated_Testing.md [hunk 2]
```diff
@@ -64,4 +64,18 @@ Integration testing, in contrast to unit testing, is meant to verify that your c
...
...s as you normally would inside Postman. Then, you can click on the "Tests" tab and create your own Postman tests here. To learn more about the syntax for doing so you can use the quick help feature which describes how to write these tests in more detail. To set up automated testing you can then right-click your API and select "Run collection". In the sidebar that opens on the right, you will now see the 3 options discussed above. Click on "Automate runs via CLI" (although it can be a good idea to run your tests manually once to confirm they work as a sanity check) and then press "Configure command" under the "Run on CI/CD" header. This will open a new page that allows you to select your CI/CD provider and the operating system environment for the pipeline. You can copy and paste the generated commands into your configuration file. Make sure you generate an API key and add it to your repository's secrets. For reference on how to do this in GitHub, see [https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions). You now have automated API testing set up through the Postman CLI!
```
[the remark this answers is not in the document]
Comment (anudevgill) [reply Topics/Development_Process/Automated_Testing.md:79]: I thought of doing this but my main concern was that so many bullet points would become overwhelming and a student may not try to implement this and of course I don't want to scare people away from trying this in their own pipelines!

[the remark this answers is not in the document]
Comment (chenikabukes) [reply Topics/Development_Process/Automated_Testing.md:79]: > I thought of doing this but my main concern was that so many bullet points would become overwhelming and a student may not try to implement this and of course I don't want to scare people away from trying this in their own pipelines!

That is a good point, Anudev. It is a clear and easy read as is, too many bullet points definitely does not seem practical for legibility purposes.

**Pull request merged**